vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -141,16 +141,20 @@ void AdsImpl::InitializeStep4(
 
   BLOG(INFO) << "Successfully initialized ads";
 
-  initialize_callback_(SUCCESS);
-
   is_foreground_ = ads_client_->IsForeground();
 
   ads_client_->SetIdleThreshold(kIdleThresholdInSeconds);
 
+  initialize_callback_(SUCCESS);
```

Comment:
curious, what does this replacement accomplish, was there a race condition of some kind before?

Reply:
There was a potential for a race-condition as when `initialize_callback_(SUCCESS)` was called, the timer for `OnIdle` was started however we had not set the threshold for this time